python/paddle/fleet/runtime/parameter_server_runtime.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
```

Comment:
I think its' better to decouple the underlying strategy for debugging. Maybe you can implement a base class for parameter server.

the ir.public implementation should be refactored into meta_optimizer or several meta_optimizers.